File: 940-fruit-into-baskets/fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        a=set()
        left=0
        maxcount=0
        count=0
        while left<len(fruits):
            if len(a)==2:
                
                if fruits[left] in a:

                    count+=1
                elif fruits[left] not in a:
                    a.add(fruits[left])
                    first=fruits[left-1]
                    right=left
                    left-=1
                    count=0
                    while fruits[left] ==first:
                        count+=1
                        left-=1
                    count+=1
                    a.remove(fruits[left])
                    left=right
            else:
                a.add(fruits[left])
                count+=1
            left+=1
            maxcount=max(maxcount,count)
        return maxcount
                    
                

File: 940-fruit-into-baskets/fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        int l=0,r=0,n=fruits.size(),maxL=0;
        map<int,int> mpp;

        while(r<n)
        {
            mpp[fruits[r]]++;
            if(mpp.size()>2)
            {
                
                    mpp[fruits[l]]--;
                    if(mpp[fruits[l]]==0)
                    mpp.erase(fruits[l]);

                    l++;
                
            }
            if(mpp.size()<=2 ) maxL=max(maxL,r-l+1);
            r++;

        }
        return maxL;
    }
};